src/ast-traverser.util.ts

```diff
@@ -55,3 +55,22 @@ export function firstAssignmentExpressionInParentChain(
     AST_NODE_TYPES.AssignmentExpression
   );
 }
+
+export function injectDecoratorFor(node: TSESTree.Node) {
```

Comment:
<details>
<summary>:corn: Corn: Non-blocking, stylistic preference or nitpick</summary>

My feedback is categorized into one of five levels:

* :seedling: Non-blocking, question
* :corn: Corn: Non-blocking, stylistic preference or nitpick
* :egg: Egg: Non-blocking, future consideration
* :baby_chick: Chick: Non-blocking, requires future action
* :chicken: Chicken: Blocking
* :sauropod: Dinosaur: Blocking, requires immediate action

See https://www.swyx.io/writing/feedback-ladders/
</details>

Can we rename this function to suggest that it returns the `@Inject` decorator. Now the name suggests that it injects a given decorator, so I was already looking for a second parameter 😄

Reply:
Ohh, good catch, I didn't notice that 'inject' could read as a verb here haha